pennylane/_device.py

```diff
@@ -302,33 +309,69 @@ def supports_observable(self, observable):
         if isinstance(observable, str):
             return observable in self.observables
 
-        raise ValueError("The given operation must either be a pennylane.Observable class or a string.")
+        raise ValueError("The given observable must either be a pennylane.Observable class or a string.")
+
+    def get_original_operation_name(self, operation_name: str):
+        """Given an inverse operation, returns the name of the original. First, it checks if the
+        device supports inverse operations.
+
+        Args:
+            operation_name (str): name of the inverse operation
+
+        Raises:
+            DeviceError: if the device does not support the computation of inverses
+
+        Returns:
+            original_name (str): name of the original operation
+        """
+        if not self.capabilities().get("inverse_operations", False):
+            raise DeviceError("The inverse of gates are not supported on device {}".format(self.short_name))
+        return operation_name[:-len(Operation.string_for_inverse)]
 
     def check_validity(self, queue, observables):
         """Checks whether the operations and observables in queue are all supported by the device.
+        Includes checks for inverse operations.
 
         Args:
             queue (Iterable[~.operation.Operation]): quantum operation objects which are intended
                 to be applied on the device
             observables (Iterable[~.operation.Observable]): observables which are intended
                 to be evaluated on the device
+
+        Raises:
+            DeviceError: if there are operations in the queue or observables that the device does
+                not support
         """
+
         for o in queue:
-            if o.name not in self.operations:
-                raise DeviceError("Gate {} not supported on device {}".format(o.name, self.short_name))
+
+            operation_name = o.name
+
+            if o.inverse:
+                operation_name = self.get_original_operation_name(o.name)
```

Comment:
If the implementation is moved to `Operation` itself (as suggested above), then the method `get_original_operation_name` can be completely removed. In this case you would just check
```python
if not self.supports_operation(operation.original_name)
```

Reply:
Also you move the check if inverse operations are supported to this part of the code, as it is currently done with the tensor observable check.

Reply:
Yep, it's better to have it in there and not organize it into a separate function this way. Thanks!